core/ipc/twin/common/src/main/java/org/opennms/core/ipc/twin/common/AbstractTwinPublisher.java

```diff
@@ -78,7 +85,12 @@ public void publish(T obj) throws IOException {
             LOG.info("Published an object update for the session with key {}", sessionKey.toString());
             byte[] value = objectMapper.writeValueAsBytes(obj);
             objMap.put(sessionKey, value);
-            handleSinkUpdate(new TwinResponseBean(sessionKey.key, sessionKey.location, value));
+            TwinResponseBean twinResponseBean = new TwinResponseBean(sessionKey.key, sessionKey.location, value);
+            // Handle local updates for the twin.
+            if (localTwinSubscriber != null) {
```

Comment:
Looking through the code, is there any time you _wouldn't_ have a local twin subscriber? Would it be better to do `Objects.nonNull(localTwinSubscriber)` up in the constructor and assume you always have one?